tests/framework/extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -0,0 +1,58 @@
+package etcdsnapshot
+
+import (
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	steveV1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+)
+
+const (
+	ProvisioningSteveResouceType = "provisioning.cattle.io.cluster"
+)
+
+func CreateSnapshot(client *rancher.Client, clustername string, namespace string) error {
+	clusterObj, existingSteveAPIObj, err := getProvisioningClusterByName(client, clustername, namespace)
+	if err != nil {
+		return err
+	}
+
+	clusterSpec := &apisV1.ClusterSpec{}
+	err = steveV1.ConvertToK8sType(clusterObj.Spec, clusterSpec)
+	if err != nil {
+		return err
+	}
+
+	if clusterSpec.RKEConfig.ETCDSnapshotCreate != nil {
+		clusterObj.Spec.RKEConfig.ETCDSnapshotCreate = &rkev1.ETCDSnapshotCreate{
+			Generation: clusterObj.Spec.RKEConfig.ETCDSnapshotCreate.Generation + 1,
+		}
+	} else {
+		clusterObj.Spec.RKEConfig.ETCDSnapshotCreate = &rkev1.ETCDSnapshotCreate{
+			Generation: 1,
+		}
+	}
+
+	_, err = client.Steve.SteveType(clusters.ProvisioningSteveResouceType).Update(existingSteveAPIObj, clusterObj)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getProvisioningClusterByName(client *rancher.Client, clusterName string, namespace string) (*apisV1.Cluster, *steveV1.SteveAPIObject, error) {
```

Comment:
Can we remove this. Looks like a spill over from development stage.

Reply:
agreed

Reply:
oh dang. i missed. ok. thanks